File: tools/improve/tests/gemini-3-flash/maudec/py/free-tuples.py
#
#	<comment from="gemini-3-flash-preview">
#	The code has been optimized by leveraging the standard 'math' module for the pi constant and consolidating the pattern matching logic in the 'first' function. The 'first' function now uses a combined match pattern to handle different tuple lengths and variants in a single branch, which improves readability while preserving the original structural validation and semantics. Redundant parentheses in the proximity check were also removed.
#	</comment>
#

import enum
import math

class Foo(enum.Enum):
    F = 0
    G = 1
    H = 2
    Err = 3

def pi() -> float:
    return math.pi

def _u003du0060u005b_u0060u005d_i(X: float, Z: float, Y: float) -> bool:
    return abs(X - Y) < Z

def first(a1: Foo) -> float:
    match a1:
        case (Foo.F, f1, _) | (Foo.G, f1, _, _) | (Foo.H, f1, _, _, _):
            return f1

File: tools/improve/tests/gemini-3-flash/maudec/rust/free-tuples.rs
//
//	<comment from="gemini-3-flash-preview">
//	I have simplified the `first` function by grouping matching arms with the `..` pattern syntax to avoid repetition. I also derived `Copy` for the `Foo` enum to improve memory efficiency and idiomatic usage for types containing only `f64` and `bool`. The `pi` function now uses the standard library constant for precision, and redundant parentheses were removed from the comparison function.
//	</comment>
//

#[derive(Clone, Copy, PartialEq, PartialOrd, Debug)]
enum Foo {
    F(f64, f64),
    G(f64, f64, bool),
    H(f64, bool, f64, bool),
    Err,
}

fn pi() -> f64 {
    std::f64::consts::PI
}

fn _u003du0060u005b_u0060u005d_(x: f64, z: f64, y: f64) -> bool {
    (x - y).abs() < z
}

fn first(a1: &Foo) -> f64 {
    match a1 {
        Foo::F(f, ..) | Foo::G(f, ..) | Foo::H(f, ..) => *f,
        _ => panic!("case not covered"),
    }
}